### PR TITLE
[SYCL][Test] Add std::complex kernel test.

### PR DESCRIPTION
**Pull request by haonanya**
Signed-off-by: haonanya <haonan.yang@intel.com>
Add std::complex kernel test.

### CODE REVIEW

File: sycl/test/devicelib/std_complex_math_fp64_test.cpp [hunk 1]
```diff
@@ -1,354 +1,203 @@
...
+// RUN: %ACC_RUN_PLACEHOLDER %t.out
+
 #include <CL/sycl.hpp>
 #include <cassert>
+#include <cmath>
+
 #include "math_utils.hpp"
-using namespace std;
 
+using std::complex;
```
Comment (bader) [inline sycl/test/devicelib/std_complex_math_fp64_test.cpp:14]: This type is declared in the `complex` header. Please, include this header.

File: sycl/test/devicelib/std_complex_math_test.cpp [hunk 2]
```diff
@@ -1,353 +1,201 @@
...
-    });
-  });
-  }
-
-  for (size_t idx = 0; idx < 4; ++idx) {
-    assert(approx_equal_cmplx(buf_out[idx], ref_results[idx]));
-  }
-}
+#define TEST_NUM1 57
+#define TEST_NUM2 10
```
Comment (alexbatashev) [inline sycl/test/devicelib/std_complex_math_test.cpp:26]:
```suggestion
static constexpr auto TestArraySize1 = 57;
static constexpr auto TestArraySize2 = 10;
```

File: sycl/test/devicelib/std_complex_math_test.cpp [hunk 3]
```diff
@@ -1,353 +1,201 @@
...
+
+complex<float> ref1_results[TEST_NUM1] = {
```
Comment (alexbatashev) [inline sycl/test/devicelib/std_complex_math_test.cpp:28]:
```suggestion
std::array<complex<float>, TestArraySize1> ref1_results = {
```

File: sycl/test/devicelib/std_complex_math_test.cpp [hunk 4]
```diff
@@ -1,353 +1,201 @@
...
+    complex<float>(0.f, M_PI_2),
+
+    complex<float>(1.f, -4.f),
+    complex<float>(18.f, -7.f),
+    complex<float>(1.557408f, 0.f),
+    complex<float>(0.f, 0.761594f),
+    complex<float>(M_PI_2, 0.f),
+    complex<float>(M_PI_2, 0.549306f)};
+
+float ref2_results[TEST_NUM2] = {0.f, 25.f, 169.f, INFINITY, 0.f,
```
Comment (alexbatashev) [inline sycl/test/devicelib/std_complex_math_test.cpp:88]:
```suggestion
std::array<float, TestArraySize2> ref2_results = {0.f, 25.f, 169.f, INFINITY, 0.f,
```

**Comment by bader** [comment]
@haonanya, do you plan to address the comments before the merge?

**Comment by haonanya** [comment]
Hi,  @alexbatashev , I already update tests using  C++ std::array to replace C array.
Thanks for your comment.

**Pull request merged**